include/tools.h

```diff
@@ -74,18 +74,8 @@ extern struct smbconf_global global_conf;
 #define KSMBD_CONF_MAP_TO_GUEST_BAD_PASSWORD	(1 << 1)
 #define KSMBD_CONF_MAP_TO_GUEST_BAD_UID		(1 << 2)
 
-#define KSMBD_CONF_DEFAULT_NETBIOS_NAME	"KSMBD SERVER"
-#define KSMBD_CONF_DEFAULT_SERVER_STRING	"SMB SERVER"
-#define KSMBD_CONF_DEFAULT_WORK_GROUP		"WORKGROUP"
-
-#define KSMBD_CONF_DEFAULT_GUEST_ACCOUNT	"nobody"
-
-#define KSMBD_CONF_DEFAULT_SESS_CAP	1024
-#define KSMBD_CONF_DEFAULT_TCP_PORT	445
-
-#define KSMBD_CONF_FILE_MAX		10000
-
-#define KSMBD_CONF_DEFAULT_CONNECTIONS	128
+#define KSMBD_CONF_MAX_OPEN_FILES	65536	/* TODO */
+#define KSMBD_CONF_MAX_ACTIVE_SESSIONS	65536	/* TODO */
```

Comment:
@namjaejeon,

Did you find these to be sane maximum values?

Reply:
@atheik Could you please elaborate more ?

Reply:
@namjaejeon,

I added `KSMBD_CONF_MAX_OPEN_FILES` and `KSMBD_CONF_MAX_ACTIVE_SESSIONS` in commit a979b31 so that `max open files = 0` and `max active sessions = 0` behave the same as `max connections = 0`, i.e. use the maximum allowed value for the parameter. I arbitrarily chose `65536` as a maximum value for `max open files` and `max active sessions`. Since you applied the commit to `master`, do you agree that `65536` is an appropriate maximum value for `max open files` and `max active sessions`?

Reply:
Yes, I agreed. BTW, You have changed the options of ksmbd.mountd and ksmbd.adduser. If ksmbd-tools is updated(or upgraded) in some product, Is it can be break if launcher doesn't change options of ksmbd-tools ? I have agreed to change the options. Looks good to me. but we don't need to consider backward compatibility ?

Reply:
@namjaejeon,

Yes, comparing the 3.4.9 release with the changes in this PR:

In `ksmbd.mountd`:
`-c` (`--config`) is now `-C` (`--config`), and
`-u` (`--users`) is now `-P` (`--pwddb`).

In `ksmbd.adduser`:
`-a` (`--add-user`)  is now `-a` (`--add`),
`-u` (`--update-user`) is now `-u` (`--update`),
`-d` (`--del-user`)  is now `-d` (`--delete`),
`USER` is a non-option argument,
`-i` (`--import-users`) is now `-P` (`--pwddb`), and
`-c` (`--config`) is now `-C` (`--config`).

In `ksmbd.addshare`:
`-a` (`--add-share`) is now `-a` (`--add`),
`-u` (`--update-share`) is now `-u` (`--update`),
`-d` (`--del-share`) is now `-d` (`--delete`),
`SHARE` is a non-option argument,
`-o` (`--options`) is now `-o` (`--option`) and can be given multiple times,
`-P` (`--pwddb`) is a new option, and
`-c` (`--config`) is now `-C` (`--config`).

Alternatively, comparing the 3.4.9 release with the changes in this PR, excluding commit 9810c89:

In `ksmbd.addshare`:
`-o` (`--options`) is now `-o` (`--option`) and can be given multiple times, and
`-i` (`--import-users`) is a new option.

I don't know of a good way to introduce backwards compatibility with the 3.4.9 release.
If you would rather not include commit 9810c89 in the next release, then that's understandable.

Reply:
Okay, Let's see what happens...